jib-core/src/main/java/com/google/cloud/tools/jib/configuration/BuildConfiguration.java

```diff
@@ -40,94 +35,33 @@
   public static class Builder {
 
     // All the parameters below are set to their default values.
-    private Instant creationTime = Instant.EPOCH;
-    @Nullable private ImageReference baseImageReference;
-    @Nullable private String baseImageCredentialHelperName;
-    @Nullable private RegistryCredentials knownBaseRegistryCredentials;
-    @Nullable private ImageReference targetImageReference;
-    @Nullable private String targetImageCredentialHelperName;
-    @Nullable private RegistryCredentials knownTargetRegistryCredentials;
-    // TODO: Should rename to not be java-specific.
-    @Nullable private ImmutableList<String> javaArguments;
-    @Nullable private ImmutableMap<String, String> environmentMap;
-    @Nullable private ImmutableList<Port> exposedPorts;
-    private Class<? extends BuildableManifestTemplate> targetFormat = V22ManifestTemplate.class;
+    private ImageConfiguration baseImageConfiguration = new ImageConfiguration();
```

Comment:
We should probably have these be the Builder classes or be `null` to start, like how `CacheConfiguration` is `null` to start.

Reply:
Any reason why? Seems like making them `null` to start just adds an extra check later (`baseImageConfiguration.getImage()` -> `baseImageConfiguration == null ? null : baseImageConfiguration.getImage()`). Seems safer to have them default to instantiated but empty.

Reply:
I think it's mostly to keep it consistent with what we have for `CacheConfiguration`, for instance. The reason that one is nullable is because initializing one (for example, the temporary directory one) may be a costly operation and we don't want to have our configuration code be coupled with possibly-costly initialization implementations. Also, in terms of semantics, having a empty `ImageConfiguration`, for example, to start, doesn't actually mean use defaults, since defaults right now are actually provided at the plugin side. We eventually want to probably add validation for different configuration types, so for `ImageConfiguration`, it should not be instantiated without an actual `ImageReference`, and having an empty constructor would mean either there would be this code path where that validation is not performed, or that the validation would need to be duplicated in the `Builder.build` and in the constructor(s) as well.

Reply:
`null` also provides the semantics at the higher-level configuration building (`BuildConfiguration`), where it can simply validate on whether required fields were `null` instead of having to understand the details of the lower-level configuration to know which fields should be validated, or call out to a lower-level configuration's validation, which probably should be performed when it was built in the first place.